kernel/kernel-api/src/main/java/io/delta/kernel/metrics/SnapshotReport.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.metrics;
+
+import java.util.Optional;
+
+/** Defines the metadata and metrics for a snapshot construction {@link MetricsReport} */
+public interface SnapshotReport extends DeltaOperationReport {
+
+  String OPERATION_TYPE = "Snapshot";
```

Comment:
I think all children of `SnapshotReport` will have operationType of `Snapshot`, right?

I think you can just define the default here.

```java
  @Override
  default void operationType() {
    return "Snapshot";
  }
```

Reply:
I'm also curious if we would ever want to see a list (to check against, compare against, etc.) of all possible operations?

would we want this to instead be an Enum inside of DeltaOperationReport?

Reply:
I'm not sure of an exact use case right now but it seems useful enough to me